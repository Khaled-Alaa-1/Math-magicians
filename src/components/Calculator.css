.calculator {
  width: 432px;
  border: 1px solid #ccc;
  border-radius: 4px;
  box-shadow: 0 0 8px rgba(0, 0, 0, 0.2);
}

.display {
  font-size: 24px;
  font-weight: 700;
  text-align: right;
  color: white;
  padding: 25px 9px 17px 0;
  background-color: #b3b3b3;
  width: 98%;
  margin-right: 39px;
}

.buttons {
  display: flex;
  width: 100%;
  height: 400px;
  flex-direction: column;
}

.button {
  font-size: 18px;
  font-weight: 700;
  padding: 16px;
  border: 1px solid rgb(184, 184, 184);
  background-color: #e0e0e0;
  cursor: pointer;
  transition: background-color 0.2s ease-in-out;
  width: 33%;
}

.button:hover {
  background-color: #d0d0d0;
}

.row {
  width: 100%;
  height: 100%;
  display: flex;
  justify-content: space-between;
}

.button-operator {
  font-size: 18px;
  font-weight: 700;
  padding: 16px;
  border: 1px solid rgb(184, 184, 184);
  background-color: orange;
  cursor: pointer;
  transition: background-color 0.2s ease-in-out;
  width: 33%;
}

.zero {
  font-size: 18px;
  font-weight: 700;
  padding: 16px;
  border: 1px solid rgb(184, 184, 184);
  background-color: #e0e0e0;
  cursor: pointer;
  transition: background-color 0.2s ease-in-out;
  width: 300px;
}
