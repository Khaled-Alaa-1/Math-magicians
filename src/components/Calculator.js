import React, { useState } from 'react';
import './Calculator.css';
import Display from './chiled';
import calculate from '../logic/calculate';

function Calculator() {
  const [calculatorData, setCalculatorData] = useState({
    total: null,
    next: null,
    operation: null,
  });

  const handleClick = (buttonName) => {
    setCalculatorData(calculate(calculatorData, buttonName));
  };

  return (
    <div className="calculator">
      <Display value={calculatorData.next || calculatorData.total || '0'} />
      <div className="buttons">
        <div className="row">
          <button type="button" className="button" onClick={() => handleClick('AC')}>AC</button>
          <button type="button" className="button" onClick={() => handleClick('+/-')}>+/-</button>
          <button type="button" className="button" onClick={() => handleClick('%')}>%</button>
          <button type="button" className="button-operator" onClick={() => handleClick('รท')}>รท</button>
        </div>
        <div className="row">
          <button type="button" className="button" onClick={() => handleClick('7')}>7</button>
          <button type="button" className="button" onClick={() => handleClick('8')}>8</button>
          <button type="button" className="button" onClick={() => handleClick('9')}>9</button>
          <button type="button" className="button-operator" onClick={() => handleClick('x')}>ร</button>
        </div>
        <div className="row">
          <button type="button" className="button" onClick={() => handleClick('4')}>4</button>
          <button type="button" className="button" onClick={() => handleClick('5')}>5</button>
          <button type="button" className="button" onClick={() => handleClick('6')}>6</button>
          <button type="button" className="button-operator" onClick={() => handleClick('-')}>-</button>
        </div>
        <div className="row">
          <button type="button" className="button" onClick={() => handleClick('1')}>1</button>
          <button type="button" className="button" onClick={() => handleClick('2')}>2</button>
          <button type="button" className="button" onClick={() => handleClick('3')}>3</button>
          <button type="button" className="button-operator" onClick={() => handleClick('+')}>+</button>
        </div>
        <div className="row">
          <button type="button" className="zero" onClick={() => handleClick('0')}>0</button>
          <button type="button" className="button" onClick={() => handleClick('.')}>.</button>
          <button type="button" className="button-operator" onClick={() => handleClick('=')}>=</button>
        </div>
      </div>
    </div>
  );
}

export default Calculator;
