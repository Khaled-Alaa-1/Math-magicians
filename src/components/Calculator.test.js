import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import { prettyDOM } from '@testing-library/dom';
import Calculator from './Calculator';

describe('Calculator', () => {
  test('calculator is rendering', () => {
    const calculator = render(<Calculator />);
    expect(prettyDOM(calculator.container)).toMatchSnapshot();
  });

  test('clicking on the "7" button should update the display', () => {
    const calculator = render(<Calculator />);
    const buttonSeven = calculator.getByText('7');
    fireEvent.click(buttonSeven);
    const displayValue = calculator.queryAllByText('7');
    expect(displayValue.length).toBeGreaterThan(0);
    expect(displayValue[0]).toBeInTheDocument();
  });
});
