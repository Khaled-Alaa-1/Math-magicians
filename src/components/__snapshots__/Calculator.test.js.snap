// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calculator calculator is rendering 1`] = `
"[36m<div>[39m
  [36m<div[39m
    [33mclass[39m=[32m\\"calculator\\"[39m
  [36m>[39m
    [36m<div[39m
      [33mclass[39m=[32m\\"display\\"[39m
    [36m>[39m
      [0m0[0m
    [36m</div>[39m
    [36m<div[39m
      [33mclass[39m=[32m\\"buttons\\"[39m
    [36m>[39m
      [36m<div[39m
        [33mclass[39m=[32m\\"row\\"[39m
      [36m>[39m
        [36m<button[39m
          [33mclass[39m=[32m\\"button AC\\"[39m
          [33mtype[39m=[32m\\"button\\"[39m
        [36m>[39m
          [0mAC[0m
        [36m</button>[39m
        [36m<button[39m
          [33mclass[39m=[32m\\"button\\"[39m
          [33mtype[39m=[32m\\"button\\"[39m
        [36m>[39m
          [0m+/-[0m
        [36m</button>[39m
        [36m<button[39m
          [33mclass[39m=[32m\\"button\\"[39m
          [33mtype[39m=[32m\\"button\\"[39m
        [36m>[39m
          [0m%[0m
        [36m</button>[39m
        [36m<button[39m
          [33mclass[39m=[32m\\"button-operator\\"[39m
          [33mtype[39m=[32m\\"button\\"[39m
        [36m>[39m
          [0m√∑[0m
        [36m</button>[39m
      [36m</div>[39m
      [36m<div[39m
        [33mclass[39m=[32m\\"row\\"[39m
      [36m>[39m
        [36m<button[39m
          [33mclass[39m=[32m\\"button\\"[39m
          [33mtype[39m=[32m\\"button\\"[39m
        [36m>[39m
          [0m7[0m
        [36m</button>[39m
        [36m<button[39m
          [33mclass[39m=[32m\\"button\\"[39m
          [33mtype[39m=[32m\\"button\\"[39m
        [36m>[39m
          [0m8[0m
        [36m</button>[39m
        [36m<button[39m
          [33mclass[39m=[32m\\"button\\"[39m
          [33mtype[39m=[32m\\"button\\"[39m
        [36m>[39m
          [0m9[0m
        [36m</button>[39m
        [36m<button[39m
          [33mclass[39m=[32m\\"button-operator\\"[39m
          [33mtype[39m=[32m\\"button\\"[39m
        [36m>[39m
          [0m√ó[0m
        [36m</button>[39m
      [36m</div>[39m
      [36m<div[39m
        [33mclass[39m=[32m\\"row\\"[39m
      [36m>[39m
        [36m<button[39m
          [33mclass[39m=[32m\\"button\\"[39m
          [33mtype[39m=[32m\\"button\\"[39m
        [36m>[39m
          [0m4[0m
        [36m</button>[39m
        [36m<button[39m
          [33mclass[39m=[32m\\"button\\"[39m
          [33mtype[39m=[32m\\"button\\"[39m
        [36m>[39m
          [0m5[0m
        [36m</button>[39m
        [36m<button[39m
          [33mclass[39m=[32m\\"button\\"[39m
          [33mtype[39m=[32m\\"button\\"[39m
        [36m>[39m
          [0m6[0m
        [36m</button>[39m
        [36m<button[39m
          [33mclass[39m=[32m\\"button-operator\\"[39m
          [33mtype[39m=[32m\\"button\\"[39m
        [36m>[39m
          [0m-[0m
        [36m</button>[39m
      [36m</div>[39m
      [36m<div[39m
        [33mclass[39m=[32m\\"row\\"[39m
      [36m>[39m
        [36m<button[39m
          [33mclass[39m=[32m\\"button\\"[39m
          [33mtype[39m=[32m\\"button\\"[39m
        [36m>[39m
          [0m1[0m
        [36m</button>[39m
        [36m<button[39m
          [33mclass[39m=[32m\\"button\\"[39m
          [33mtype[39m=[32m\\"button\\"[39m
        [36m>[39m
          [0m2[0m
        [36m</button>[39m
        [36m<button[39m
          [33mclass[39m=[32m\\"button\\"[39m
          [33mtype[39m=[32m\\"button\\"[39m
        [36m>[39m
          [0m3[0m
        [36m</button>[39m
        [36m<button[39m
          [33mclass[39m=[32m\\"button-operator\\"[39m
          [33mtype[39m=[32m\\"button\\"[39m
        [36m>[39m
          [0m+[0m
        [36m</button>[39m
      [36m</div>[39m
      [36m<div[39m
        [33mclass[39m=[32m\\"row\\"[39m
      [36m>[39m
        [36m<button[39m
          [33mclass[39m=[32m\\"zero\\"[39m
          [33mtype[39m=[32m\\"button\\"[39m
        [36m>[39m
          [0m0[0m
        [36m</button>[39m
        [36m<button[39m
          [33mclass[39m=[32m\\"button\\"[39m
          [33mtype[39m=[32m\\"button\\"[39m
        [36m>[39m
          [0m.[0m
        [36m</button>[39m
        [36m<button[39m
          [33mclass[39m=[32m\\"button-operator\\"[39m
          [33mtype[39m=[32m\\"button\\"[39m
        [36m>[39m
          [0m=[0m
        [36m</button>[39m
      [36m</div>[39m
    [36m</div>[39m
  [36m</div>[39m
[36m</div>[39m"
`;
